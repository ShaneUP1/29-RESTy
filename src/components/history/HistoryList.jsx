import React from 'react';
import PropTypes from 'prop-types';
import SingleHistory from './SingleHistory';
// import styles from './HistoryList.css';

const HistoryList = ({ histories }) => {
  const historyList = histories.map((history, i) => (
    <li key={i}>
      <SingleHistory {...history}/>
    </li>
  ));

  return (
    <ul>
      {historyList}
    </ul>
  );
};

HistoryList.propTypes = {
  histories: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default HistoryList;

