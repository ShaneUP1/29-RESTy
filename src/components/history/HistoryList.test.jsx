import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList';


describe('HistoryList component', () => {
  afterEach(() => cleanup());
  it('renders a list of histories', () => {
    const histories = [{ url: 'www.url.com', method: 'GET' }, { url: 'www.url.com', method: 'GET' }, { url: 'www.url.com', method: 'GET' }];
    const { asFragment } = render(<HistoryList
      histories={histories}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
