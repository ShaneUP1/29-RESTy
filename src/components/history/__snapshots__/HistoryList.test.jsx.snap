// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders a list of histories 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <div
        class="SingleHistory"
      >
        <p>
          www.url.com
        </p>
        <p>
          GET
        </p>
      </div>
    </li>
    <li>
      <div
        class="SingleHistory"
      >
        <p>
          www.url.com
        </p>
        <p>
          GET
        </p>
      </div>
    </li>
    <li>
      <div
        class="SingleHistory"
      >
        <p>
          www.url.com
        </p>
        <p>
          GET
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
